src/compiler/checker.ts

```diff
@@ -4450,6 +4518,10 @@ namespace ts {
             return type.resolvedApparentType;
         }
 
+        function getApparentTypeOfSpread(type: SpreadType) {
+            return getApparentType(type.right);

```

Comment:
Can you explain the intuition here? Shouldn't the apparent type be the flattened object type that right side properties overlaid on left-side properties?